src/core/shape_inference/include/embedding_shape_infer_utils.hpp

```diff
@@ -0,0 +1,44 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+#pragma once
+
+#include "openvino/core/node.hpp"
+
+namespace ov {
+namespace op {
+namespace util {
+namespace embedding {
+/**
+ * \brief Return a copy of the source shape with the first dimension replaced by provided dim
+ *
+ * \tparam TShape        Shape type
+ * \tparam TDim          Dimension type
+ *
+ * \param op             Pointer to operator.
+ * \param shape_src      The shape to be copied
+ * \param dim            The replacement dimension to overwirite the first one
```

Comment:
It can be named as `batch_shape` as it may have batch dimension required for output shape.

Reply:
The dimension is called "num_segments" for [EmbeddingSegmentsSum_3](https://github.com/openvinotoolkit/openvino/blob/master/docs/ops/sparse/EmbeddingSegmentsSum_3.md), so I decided to keep this name more generic.